### ISSUE TITLE
Train on my custom dataset, mAP = 0

### ISSUE THREAD

**Issue by sakurasakura1996** [eriklindernoren/PyTorch-YOLOv3#468]
use  command  !CUDA_VISIBLE_DEVICES=0 python3 train.py --epochs 50 --model_def config/yolov3-custom.cfg --data_config config/custom.data --pretrained_weights weights/darknet53.conv.74 to train my dataset.after first epoch, Console shewed:
Computing AP: 100% 5/5 
+--------+--------------+-------------+
|   Index  |  Class name  |  AP              |
+--------+--------------+-------------+
|  0          |  xxxxx           |  0.0000000   |
|  1          |  xxxxx           |  0.0000000   |
|  2          |  xxxxx           |  0.0000000   |
|  3          |  xxxxx           |  0.0000000   |
|  5          |  xxxxx           |  0.0000000   |
+--------+--------------+-------------+
----mAP 0.0
Anyone have met this problem and solved it?

**Comment by xycim** [comment]
hi，生成lable的代码可以分享我一下吗？我现在只有jpg图片和对应的xml，下一步不会做了。感谢。

**Comment by sakurasakura1996** [comment]
你好，我在训练自定义的数据集之后，预测时发现完全是错的，估计可能就是这里的坐标转换除了问题，今天正好在看，代码是下面的，你可以参考方法，但是从结果来看，转换方式应该是有问题的。
import os
import os.path
import xml.dom.minidom


path = "C:/Users/Administrator/Desktop/AI_studio/PaddleDetection/dataset/insect/Annotations"
write_path = "C:/Users/Administrator/Desktop/pytorch-yolov3/custom/labels"
files = os.listdir(path)  # 得到文件夹下所有文件名称

label_dict = {
    'Boerner':0,
    'linnaeus':1,
    'armandi':2,
    'coleoptera':3,
    'Linnaeus':4,
    'Leconte':5,
    'acuminatus':6
}
for filename in files:
    dom = xml.dom.minidom.parse(os.path.join(path,'1.xml'))
    root = dom.documentElement
    width = int(root.getElementsByTagName('width')[0].firstChild.data)
    height = int(root.getElementsByTagName('height')[0].firstChild.data)
    name = root.getElementsByTagName('name')
    xmin = root.getElementsByTagName('xmin')
    ymin = root.getElementsByTagName('ymin')
    xmax = root.getElementsByTagName('xmax')
    ymax = root.getElementsByTagName('ymax')
    # for i in range(len(xmin)):
    #     print(xmin[i].firstChild.data)
    #     print(ymin[i].firstChild.data)
    #     print(xmax[i].firstChild.data)
    #     print(ymax[i].firstChild.data)


    with open(os.path.join(write_path,filename.split('.')[0]+'.'+'txt'),'w',encoding='UTF-8') as fh:
    # 然后根据每一个框的上面五个信息，计算出格式为  index(目标框所属分类的索引）,中心点横坐标，中心点纵坐标，宽，高
        for i in range(len(name)):
            idx = label_dict[name[i].firstChild.data]
            center_x = round((int(xmin[i].firstChild.data) + (int(xmax[i].firstChild.data))/2)/width,3)
            center_y = round((int(ymin[i].firstChild.data) + (int(ymax[i].firstChild.data))/2)/height,3)
            width_gt = round((int(xmax[i].firstChild.data)-int(xmin[i].firstChild.data))/width,8)
            height_gt = round((int(ymax[i].firstChild.data)-int(ymin[i].firstChild.data))/height, 8)

            print(idx,center_x,center_y,width_gt,height_gt)
            str_temp = str(idx) +' ' + str(center_x) +' ' + str(center_y) +' ' + str(width_gt) +' ' + str(height_gt)+ '\n'
            fh.write(str_temp)
    fh.close()




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"xycim"<notifications@github.com&gt;;
发送时间:&nbsp;2020年4月2日(星期四) 下午4:54
收件人:&nbsp;"eriklindernoren/PyTorch-YOLOv3"<PyTorch-YOLOv3@noreply.github.com&gt;;
抄送:&nbsp;"★夜尽天明☆™"<2470375551@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [eriklindernoren/PyTorch-YOLOv3] Train on my custom dataset, mAP = 0 (#468)





 
hi，生成lable的代码可以分享我一下吗？我现在只有jpg图片和对应的xml，下一步不会做了。感谢。

**Comment by CRRRRR2016** [comment]
你好，我想问一下你生成的x,y,w,h有进行归一化处理吗，我也遇到了mAP=0的问题

**Comment by sakurasakura1996** [comment]
我进行了归一化处理，但是可能不对。我已经换了一个yolo仓库。



---Original---

**Comment by vipulkumar-developer** [comment]
Can someone translate this conversation?

**Comment by Flova** [comment]
Is this issue solved?

**Comment by caihaunqai** [comment]
> 我进行了归一化处理，但是可能不对。我已经换了一个yolo仓库。
> […](#)
> ---Original--- From: "CRRRRR2016"<notifications@github.com&gt; Date: Mon, Apr 13, 2020 12:18 PM To: "eriklindernoren/PyTorch-YOLOv3"<PyTorch-YOLOv3@noreply.github.com&gt;; Cc: "Author"<author@noreply.github.com&gt;;"sakura Chen"<2470375551@qq.com&gt;; Subject: Re: [eriklindernoren/PyTorch-YOLOv3] Train on my custom dataset, mAP = 0 (#468) 你好，我想问一下你生成的x,y,w,h有进行归一化处理吗，我也遇到了mAP=0的问题 — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

您好，我也是跑自己的数据集，MAP0，我觉得数据处理应该没问题，可以加一下联系方式，请教您一下吗？

**Comment by unw9527** [comment]
This is not solved. I trained on my own dataset, and after validation in every epoch, map is 0.
